x-pack/plugins/maps/public/shared/components/layer_toc_actions.js

```diff
@@ -52,73 +51,89 @@ export class LayerTocActions extends Component {
     }));
   };
 
-  _renderButton() {
-    const icon = this._renderIcon();
+  _renderPopoverToggleButton() {
+    const { icon, tooltipContent } = this._getIconAndTooltipContent();
     return (
-      <EuiButtonEmpty
-        className="mapTocEntry__layerName eui-textLeft"
-        size="xs"
-        flush="left"
-        color="text"
-        onClick={this._onClick}
-        data-test-subj={`layerTocActionsPanelToggleButton${this.props.escapedDisplayName}`}
-        // textProps="mapTocEntry__layerNameText"
+      <EuiToolTip
+        position="top"
+        title={this.props.displayName}
+        content={tooltipContent}
       >
-        <span className="mapTocEntry__layerNameIcon">{icon}</span>
-        {this.props.displayName}
-      </EuiButtonEmpty>);
-  }
-
-
-  _getVisbilityIcon() {
-    const iconType = this.props.layer.isVisible() ? 'eye' : 'eyeClosed';
-    return (
-      <EuiIcon
-        type={iconType}
-        size="m"
-      />);
+        <EuiButtonEmpty
+          className="mapTocEntry__layerName eui-textLeft"
+          size="xs"
+          flush="left"
+          color="text"
+          onClick={this._togglePopover}
+          data-test-subj={`layerTocActionsPanelToggleButton${this.props.escapedDisplayName}`}
+          // textProps="mapTocEntry__layerNameText"
+        >
+          <span className="mapTocEntry__layerNameIcon">{icon}</span>
+          {this.props.displayName}
+        </EuiButtonEmpty>
+      </EuiToolTip>
+    );
   }
 
-  _renderIcon() {
+  _getIconAndTooltipContent() {
     const { zoom, layer } = this.props;
-    let smallLegendIcon;
+
     if (layer.hasErrors()) {
```

Comment:
fwiw, vanilla if-elseif-else is more readable than returning early in all these floating ifs. It shows much clearer all branches are mutually exclusive, and the `return` statements aren't sprinkled all throughout the message body.

Early returns work well imho when dealing with some non-default case really quickly early in a function, but not when the branching is core to the logic of the routine.